### PR TITLE
implement unmix_intervals

### LINKED ISSUE #96: implement forecast.unmix_intervals
Tests in #95 only pass because `forecast.unmix_intervals` is commented out. Need to implement this for GFS forecasts.

### LINKED ISSUE #104: implement gfs_quarter_deg_to_hourly_mean
The `reference_forecasts.models.gfs_quarter_deg_to_hourly_mean` function is supposed to be able to handle the switch from hourly to > hourly output.

### PR DESCRIPTION
**Pull request by wholmgren**
closes #96 

closes #104

### CODE REVIEW

File: solarforecastarbiter/reference_forecasts/forecast.py [hunk 1]
```diff
@@ -213,10 +216,40 @@ def unmix_intervals(cloud_cover):
...
+        Data is the unmixed interval average with ending label.
     """
-    raise NotImplementedError
+    intervals = (mixed.index[1:] - mixed.index[:-1]).unique()
+    if len(intervals) > 1:
+        raise ValueError('multiple interval lengths detected. slice forecasts '
+                         'into sections with unique interval lengths first.')
+    interval = intervals[0]
+    mixed_vals = np.array(mixed)
+    if interval == pd.Timedelta('1h'):
```
Comment (alorenzo175) [inline solarforecastarbiter/reference_forecasts/forecast.py]: I think some comments here about exactly what this will be helpful

Comment (wholmgren) [reply solarforecastarbiter/reference_forecasts/forecast.py]: is it not clear from the doc string?

Comment (alorenzo175) [reply solarforecastarbiter/reference_forecasts/forecast.py]: It just took me a few minutes to understand what mixed intervals are and that your math is right. An inline comment like 
``mixed_3 = avg 0-3 = (0-1 + 1-2 + 1-3) /3 => avg 1-3 = 3 * mixed3 - (0-1 + 1-2) = 3 * mixed_3 - 2 * mixed_2``
would help I think

Comment (wholmgren) [reply solarforecastarbiter/reference_forecasts/forecast.py]: hopefully makes more sense now.

Comment (alorenzo175) [reply solarforecastarbiter/reference_forecasts/forecast.py]: yes

**Comment by wholmgren** [comment]
@alorenzo175 I decided it made sense to implement `gfs_quarter_deg_to_hourly_mean` in this PR because it has to deal with changes in mixed intervals. Might be nice if `unmixed_intervals` could handle multiple interval lengths in the future, but I think this is good enough for now. Anyways, it needs another review.

**Pull request merged**